metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/WorkflowUtils.java

```diff
@@ -102,41 +111,58 @@ public WorkflowUtils(WorkflowExecutionDao workflowExecutionDao) {
       throw new BadContentException("Workflow should not be empty.");
     }
 
-    // Compile the list of enabled plugins.
-    final List<AbstractExecutablePluginMetadata> plugins = workflow.getMetisPluginsMetadata()
+    // Compile the list of enabled enabledPlugins.
+    final List<AbstractExecutablePluginMetadata> enabledPlugins = workflow.getMetisPluginsMetadata()
         .stream().filter(AbstractExecutablePluginMetadata::isEnabled).collect(Collectors.toList());
 
     // Sanity checks: workflow should not be empty and all should have a type.
-    if (plugins.isEmpty()) {
+    if (enabledPlugins.isEmpty()) {
       throw new BadContentException("Workflow should not be empty.");
     }
-    if (plugins.stream().map(AbstractExecutablePluginMetadata::getExecutablePluginType)
+    if (enabledPlugins.stream().map(AbstractExecutablePluginMetadata::getExecutablePluginType)
         .anyMatch(Objects::isNull)) {
-      throw new BadContentException("There are plugins of which the type could not be determined.");
+      throw new BadContentException(
+          "There are enabledPlugins of which the type could not be determined.");
     }
 
-    if (plugins.size() > 1 && plugins.stream()
-        .map(AbstractExecutablePluginMetadata::getExecutablePluginType)
-        .anyMatch(
-            executablePluginType -> executablePluginType.toPluginType() == PluginType.DEPUBLISH)) {
-      throw new BadContentException("If DEPUBLISH plugin enabled, no other plugins are allowed.");
+    // If depublish requested, make sure it's the only plugin in the workflow
+    final Optional<DepublishPluginMetadata> depublishPluginMetadata = enabledPlugins.stream()
+        .filter(plugin -> plugin.getExecutablePluginType().toPluginType() == PluginType.DEPUBLISH)
+        .map(plugin -> (DepublishPluginMetadata) plugin).findFirst();
+    if (enabledPlugins.size() > 1 && depublishPluginMetadata.isPresent()) {
+      throw new BadContentException(
+          "If DEPUBLISH plugin enabled, no other enabledPlugins are allowed.");
     }
 
-    // Validate and normalize the harvest parameters of harvest plugins (even if not enabled)
-    validateAndTrimHarvestParameters(plugins);
+    // If record depublication requested, check if there are pending record ids in the db
+    if (depublishPluginMetadata.isPresent() && !depublishPluginMetadata.get()
+        .isDatasetDepublish()) {
+      final Set<String> pendingDepublicationIds = depublishRecordIdDao
+          .getAllDepublishRecordIdsWithStatus(workflow.getDatasetId(),
+              DepublishedRecordSortField.DEPUBLICATION_STATE, SortDirection.ASCENDING,
+              DepublicationStatus.PENDING_DEPUBLICATION);
+      if (CollectionUtils.isEmpty(pendingDepublicationIds)) {
+        throw new BadContentException(
+            "Record depublication requested but there are no pending depublication record ids in the db");
+      }
+    }
+
```

Comment:
I think adding this code put the cognitive complexity of this method over the line.

Reply:
Reduced it